main.cpp

```diff
@@ -1,48 +1,557 @@
 /*
- Project 4 - Part 1 / 9
- video: Chapter 2 Part 7
- Returning Values tasks 
+ Project 4 - Part 2 / 9
+ Video: Chapter 3 Part 6
 
- Create a branch named Part1
+ Create a branch named Part2
 
- Purpose:  This project will take you through the process of writing a class that wraps a numeric type, beginning with writing simple member functions and ending with a fully templated class with lots of functionality. 
- 
- 1) write 3 UDTs named FloatType, DoubleType, IntType.
- 
- 2) give each UDT the following member functions:
-        add( lhs, rhs );
-        subtract( lhs, rhs );
-        multiply( lhs, rhs );
-        divide( lhs, rhs );
- 3) implement the appropriate action in the member function. 
-         a) Be aware that floating point division by 0 is legal, but integer division by 0 will crash your program.  
-         b) Handle this possible input when you write your divide() functions.
-         c) you should warn the user if they're doing floating-point-division-by-zero but not prevent the division from happening
-         d) you should warn AND prevent the division from happening if it is an integer-division-by-zero.
- 
- 4) make them return the correct primitive type. e.g. if you're implementing the FloatType::add function, your implementation would start like this:
-        float FloatType::add( float lhs, float rhs )
+New/This/Pointers/References conclusion
+
+    Build/Run often with this task to make sure you're not breaking the code with each step.
+    I recommend committing after you get each step working so you can revert to a working version easily if needed.
  
- 5) declare a few instances of each class in main() and call the member functions with the appropriate arguments.
-        FloatType ft;
-        auto result = ft.add(3.2f, 23.f );
+ 0) in the blank space below, declare/define an empty struct named 'A' on a single Line. 
+     on the lines below it, write a struct named 'HeapA' that correctly shows how to own an instance of 'A' 
+         on the heap without leaking, without using smart pointers. 
+ */
+
+struct A {};
+struct HeapA
+{
+    A* pointerToA = nullptr;
+
+    HeapA( )
+    { 
+        pointerToA = new A;
+    }
```

Comment:
You should use the member initializer list here instead of inside the constructor body.